### DISCUSSION TITLE
Welcome to csharplang Discussions!

### DISCUSSION THREAD

**Discussion by MadsTorgersen** [dotnet/csharplang#3863] [General]
Hi all,
 
GitHub now allows Discussions, and we have enabled them in this repo. This is going to be great for us, as a large number of our issues are currently discussions in disguise, and will be much better served by the new feature. We’ve updated the guidance [here](https://github.com/dotnet/csharplang/blob/master/README.md#discussions) to reflect this change: Issues should now only be used for specific concrete changes that are proposed for the repo, e.g. worked-out language proposals as well as bug reports on what’s in the repo.
 
As part of transitioning we’re moving recently active issues with the Discussion label to in fact be discussions. If you feel something has been moved in error, please log a fresh issue for it! As we all get used to this change we will continue to summarily move discussion issues to discussions as we see them. Please bear with us as we find the right balance here, and let us know any ideas you have for improving how we manage the split between issues and discussions!
 
Discussions should be on topic, but we strongly encourage an explorative and supportive tone. Put on your yes-hat and be constructive when responding to other people’s ideas. Make sure your response communicates new insight, not just an emotion – which can be captured better in a reaction emoji.

Looking forward to many great discussions!

Mads

**Comment by jnm2** [comment]
> If you are reporting a bug or making a fully worked out feature proposal, please go ahead and open it as an issue. If you have a topic or a looser feature idea then please open it as a discussion.

This seems like a good thing to put in the issue template. Even better perhaps a template chooser because they can redirect you to arbitrary links like to the "New discussion" page. For example, this is what happens when you click Roslyn's "New issue" button: https://github.com/dotnet/roslyn/issues/new/choose

**Reply by 333fred** [reply]
I've submitted a PR for this: https://github.com/dotnet/csharplang/pull/3872

**Comment by MgSam** [comment]
Meh, I think using tags to differentiate between issue types filed under "Issues" is better than having a second forum which is almost certainly going to get even less attention than the primary one.  

This seems like a step backwards for Github, IMHO. Segregated forums were popular 15 years ago but are an increasing rarity these days...

**Reply by alrz** [reply]
Back in time when the team was considering GitHub vs Discourse vs mailing list, I recall they felt the need for a Discussion feature with threaded comments and whatnot. This is exactly what they had in mind.

**Reply by bbarry** [reply]
I wonder if this is nested threaded like reddit or newsgroups? guess I'll find out with this reply :)

I think there is merit to both an all in one place approach and a separate threaded discussion feature. 

It seems to me that this is a UI issue for github to solve: come up with a default mechanism to present a single discussion interface that permits some discussions to be issues and others to be PRs and others to simply be. Then provide an opt-out to segregated tabs.

I'll admit I am abusing the notifications area in my profile for this and simply watching everything in a few dozen repositories for a single place to look for anything to happen. The signal to noise ratio is very low though and I would like more fine tuned control to perhaps create rules that ignore things meeting certain criteria. Still that is an opt-in approach that I think does a disservice to everyone because it both fails to push the uninformed to learn where certain communities preferences lay and fails to provide the control that is needed to not just see everything.
